### PR TITLE
Cypress code enhancements

### PR DESCRIPTION
**Pull request by pratibhapandey16**
### Description
Updated cypress test scripts with new implementations in Pie, Treemap and Histogram charts

### Check List
- [ ] New functionality includes testing.
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

@spattnaik @abasatwar

### CODE REVIEW

File: dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/// <reference types="cypress" />
+import {
+    delay,
+    TEST_QUERIES,
+    querySearch,
+    landOnEventVisualizations,
+  } from '../../utils/event_constants';
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js:12]: check formatting of code. Use prettier.

Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js:12]: check formatting of code. Use prettier.

File: dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    it.only('Renders Histogram chart and Data Configurations panel', () => {
+     cy.get('.euiResizablePanel.euiResizablePanel--collapsible.euiResizablePanel--middle')
+       .contains('Data Configurations');
+     cy.get('.euiResizablePanel.euiResizablePanel--collapsible.euiResizablePanel--middle')
+       .contains('Bucket Size');
+     cy.get('[data-test-subj="valueFieldNumber"]').eq(0).type('500');
+     cy.get('[data-test-subj="valueFieldNumber"]').eq(1).type('1000');
+     cy.get('.euiPanel.euiPanel--paddingSmall.euiPanel--borderRadiusMedium.euiPanel--plain.euiPanel--hasShadow.dataConfigContainer').click();   
+    });
+  });
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js:61]: add empty line.

File: dashboards-observability/.cypress/integration/VisualizationCharts/7_pie_chart.spec.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      cy.get('[data-text="Show"] [data-test-subj="show"]').should('have.attr', 'checked');
+      cy.get('[data-text="Hidden"]').should('have.text', 'Hidden').click();
+      cy.get('[data-text="Hidden"] [data-test-subj="hidden"]').should('not.have.attr', 'checked');
+      cy.get('[data-test-subj="visualizeEditorRenderButton"]').click({ force: true,multiple: true });
+    });
+  
+    it.only('Renders Pie chart with single color', () => {
+      renderPieChart();
+    });
+  });
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/7_pie_chart.spec.js:117]: same as above

Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/7_pie_chart.spec.js:117]: same as above

File: dashboards-observability/.cypress/integration/VisualizationCharts/8_treemap_chart.spec.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+/// <reference types="cypress" />
+import {
+  delay,
+  TEST_QUERIES,
+  querySearch,
+  landOnEventVisualizations,
+  renderDataConfig,
+  saveVisualizationAndVerify,
+  deleteVisualization,
+} from '../../utils/event_constants';
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/8_treemap_chart.spec.js:15]: check formatting of all files and lint errors too.

File: dashboards-observability/.cypress/integration/VisualizationCharts/8_treemap_chart.spec.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    cy.get('.euiTitle.euiTitle--xxsmall').eq(1).contains('Dimensions');
+    cy.get('.first-division .euiFormLabel.euiFormRow__label').eq(0).contains('Child Field');
+    cy.get('.first-division [data-test-subj="comboBoxInput"]').eq(0).click({ force: true });
+    cy.get('.euiComboBoxOption__content').eq(2).click();
+    renderAddParent();
+    cy.get(' [data-test-subj="addParentButton"] .euiButton__text').contains('+ Add Parent').click();
+    cy.get('.first-division .euiFormLabel.euiFormRow__label').eq(4).contains('Parent 4').should('exist');
+    cy.get('.euiTextColor.euiTextColor--subdued').contains('No results found').should('exist');
+  });
+});
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/8_treemap_chart.spec.js:194]: same as above

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/.cypress/utils/event_constants.js [hunk 6]
```diff
@@ -3,42 +3,54 @@
...
-    dateRangeDOM: YEAR_TO_DATE_DOM_ID
+    query:
+      'source=opensearch_dashboards_sample_data_flights | stats max(AvgTicketPrice) by DestCountry, DestCityName, Carrier',
+    dateRangeDOM: YEAR_TO_DATE_DOM_ID,
   },
   {
-    query: 'source = opensearch_dashboards_sample_data_logs | stats count(), avg(bytes) by host, tags',
-    dateRangeDOM: YEAR_TO_DATE_DOM_ID
+    query:
+      'source = opensearch_dashboards_sample_data_logs | stats count(), avg(bytes) by host, tags',
```
Comment (abasatwar) [inline dashboards-observability/.cypress/utils/event_constants.js:32]: check formatting of file and update the code as per prettier.

File: dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  it('Renders Histogram chart and Color Theme', () => {});
+
+  it.only('Renders Histogram chart and Data Configurations panel', () => {
+    cy.get('.euiResizablePanel.euiResizablePanel--collapsible.euiResizablePanel--middle').contains(
+      'Data Configurations'
+    );
+    cy.get('.euiResizablePanel.euiResizablePanel--collapsible.euiResizablePanel--middle').contains(
+      'Bucket Size'
+    );
+    cy.get('[data-test-subj="valueFieldNumber"]').eq(0).type('500');
```
Comment (abasatwar) [inline dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js:63]: use constants instead of magic numbers.

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/.cypress/utils/event_constants.js (hunk 6 above)
Comment (pratibhapandey16) [reply dashboards-observability/.cypress/utils/event_constants.js:32]: The code is already updated using preitter.

**Review by abasatwar — APPROVED** [review]
looks good.

File: dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js (hunk 7 above)
Comment (pratibhapandey16) [reply dashboards-observability/.cypress/integration/VisualizationCharts/5_histogram_chart.spec.js]: sure

**Review by spattnaik — APPROVED** [review]

**Comment by pratibhapandey16** [comment]
@mengweieric, could you please merge these changes in main?

**Comment by anirudha** [comment]
closing older test PR with conflict.

**Pull request closed without merging**